### DISCUSSION TITLE
Unable to run binaryninja: qt.qpa.xcb: could not connect to display 127.0.0.1:0.0

### DISCUSSION THREAD

**Discussion by Ewael** [Vector35/binaryninja-api#1753] [General] [answered]
Hello,

I just bought the license for Binary Ninja but I can't run it. I sent a mail to the support but no response. Here is the problem.

After unzipping the downloaded file I installed it with `bash scripts/linux-setup.sh`. But now when trying to run it I have:

```
No protocol specified
qt.qpa.xcb: could not connect to display :0.0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Aborted
```

Turning the debug mode on with `export QT_DEBUG_PLUGINS=1`, I have:

```
QFactoryLoader::QFactoryLoader() checking directory path "/opt/binaryninja/qt/platforms" ...
QFactoryLoader::QFactoryLoader() looking at "/opt/binaryninja/qt/platforms/libqxcb.so"
Found metadata in lib /opt/binaryninja/qt/platforms/libqxcb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "xcb"
        ]
    },
    "archreq": 0,
    "className": "QXcbIntegrationPlugin",
    "debug": false,
    "version": 331264
}


Got keys from plugin meta data ("xcb")
QFactoryLoader::QFactoryLoader() checking directory path "/opt/binaryninja/platforms" ...
loaded library "/opt/binaryninja/qt/platforms/libqxcb.so"
qt.qpa.xcb: could not connect to display 127.0.0.1:0.0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Aborted
```

I've already tried to install some libs that could be missing after some googling but no luck. I've also added `export DISPLAY=127.0.0.1:0.0` in my bashrc.

Here is the output of `uname -a`:

```
Linux kali 5.6.0-kali2-amd64 #1 SMP Debian 5.6.14-1kali1 (2020-05-25) x86_64 GNU/Linux
```

Thank's for reading! If you need any information it would be my pleasure to give them!

**Comment by negasora** [comment]
Similar errors are usually due to the system missing the equivalent of the `libxcb` or `libxcb-dev` packages. However, the fact that the default display can't be found suggests another problem.
Are you able to run any Qt applications?

**Reply by Ewael** [reply]
I do not find `libxdb` or `libxcb-dev`. I have the following:
```
libxcb1:amd64               libxcb-keysyms1:amd64       libxcb-shape0:amd64         libxcb-xinerama0:amd64    
libxcb1-dev:amd64           libxcb-present0:amd64       libxcb-shape0-dev:amd64     libxcb-xinerama0-dev:amd64
libxcb-dri2-0:amd64         libxcb-randr0:amd64         libxcb-shm0:amd64           libxcb-xinput0:amd64      
libxcb-dri3-0:amd64         libxcb-randr0-dev:amd64     libxcb-shm0-dev:amd64       libxcb-xkb1:amd64         
libxcb-glx0:amd64           libxcb-render0:amd64        libxcb-sync1:amd64          libxcb-xkb-dev:amd64      
libxcb-icccm4:amd64         libxcb-render0-dev:amd64    libxcb-util0:amd64          libxcb-xtest0:amd64       
libxcb-image0:amd64         libxcb-render-util0:amd64   libxcb-xfixes0:amd64        libxcb-xtest0-dev:amd64 
```

And could you please provide an example of a QT application that I could try to test?

**Reply by negasora** [reply]
I think `wireshark` uses the Qt frontend on kali. It's installed by default.

**Reply by Ewael** [reply]
`wireshark` works really well, never noticed any problems on it.

**Comment by Ewael** [comment, marked as answer]
Problem solved, this was related to xauth which was not correctly set for `kali` user.

**Reply by negasora** [reply]
`xhost +` was the solution. Note that this will grant all global access to the xserver, so something like `xhost +SI:localuser:<username>` is a better solution.

**Reply by Ewael** [reply]
Nah `xhost +` was not working, I had to add `export XAUTHORITY=~/.Xauthority` in the `kali` .bashrc because it was trying to take the root one by default thus there was a rights problem indeed. I guess that was my solution / problem but you are probably right there too ahah

**Comment by psifertex** [comment]
To be clear -- the better solution is just don't login as root in the first place.
